stdlib/seq.mc

```diff
@@ -472,16 +472,59 @@ utest filterOption [Some 3, Some 2, None (), Some 4] with [3, 2, 4] using eqSeq
 utest filterOption [None (), None ()] with [] using eqSeq eqi
 utest filterOption [None (), Some 1, None (), Some 1] with [1, 1] using eqSeq eqi
 
+-- Find the first element in a sequence satisfying a predicate in a list.
 recursive
   let find : all a. (a -> Bool) -> [a] -> Option a = lam p. lam seq.
     if null seq then None ()
     else if p (head seq) then Some (head seq)
     else find p (tail seq)
 end
 
+let eqRec = lam lhs. lam rhs.and (eqi lhs.x rhs.x) (eqi lhs.y rhs.y)
```

Comment:
This adds to the global namespace, which might not be ideal. What does this test case capture that the others do not?